src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+        }
+        if (fabsf(newPos.z) < 10.0f) {
+            newPos.z = 0.0;
+        }
+
+        if ((newPos.x == 0.0f) && (newPos.z == 0.0f)) {
+            this->unkTimer250 = 0;
+            this->unk298 = 7;
+        }
+
+        Math_SmoothStepToS(&this->actor.world.rot.y, Math_Atan2S(newPos.x, newPos.z), 3, this->unk304, 0);
+        Math_ApproachF(&this->unk304, 10000.0f, 1.0f, 1000.0f);
+    }
+    func_80891320(this, globalCtx, s16tmp);
+}
+#else
 #pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Niw_0x80891060/func_808919E8.asm")
+#endif
+
+void EnNiw_Held(EnNiw* this, GlobalContext* globalCtx) {
+    Vec3f vec3fcopy;
+    s16 rotZ;
+
+    // again with the double copy?
+    vec3fcopy = D_808934DC;
+    if (this->unkTimer250 == 0) {
+        this->unk29E = 2;
+        // cast req
+        this->unkTimer250 = (s32)(Rand_ZeroFloat(1.0f) * 10.0f) + 10;
+    }
+
+    // casts req
+    this->actor.shape.rot.x = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.x;
+    this->actor.shape.rot.y = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.y;
+    this->actor.shape.rot.z = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.z;
+    if (this->niwType == ENNIW_TYPE_REGULAR) {
+        if (Actor_HasNoParent(&this->actor, globalCtx)) {
+            this->actor.shape.rot.z = 0;
+            rotZ = this->actor.shape.rot.z;
+            this->unknownState28E = 5;
+            this->actor.flags |= 0x1; // targetable ON
+            this->actionFunc = EnNiw_Thrown;
+            this->actor.shape.rot.y = rotZ;
+            this->actor.shape.rot.x = rotZ;
+        }
+
+    } else if (this->unk2BC.z != 0.0f) {
+        // if not regular..? when does this go off?
+        this->actor.shape.rot.z = 0;
+        rotZ = this->actor.shape.rot.z;
+        this->actor.velocity.y = 8.0f;
+        this->actor.speedXZ = 4.0f;
+        this->actor.gravity = -2.0f;
+        this->unknownState28E = 5;
+        this->unk2EC = 0;
+        this->niwType = ENNIW_TYPE_REGULAR;
+        this->actor.shape.rot.y = rotZ;
+        this->actor.shape.rot.x = rotZ;
+         Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor,  &sCylinderInit );
+        Math_Vec3f_Copy(&this->unk2BC, &vec3fcopy);
+        this->actor.flags |= 0x1; // targetable ON
+        this->actionFunc = EnNiw_Thrown;
+    }
+    func_80891320(this, globalCtx, 2);
+}
+
+// action function: recently thrown, and also hopping on the floor
+void EnNiw_Thrown(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->unk2EC == 0) {
+        if ((this->actor.bgCheckFlags & 1)) { // about to hit the floor
+            this->unk2EC = 1;
+            this->unkTimer252 = 80; // hop timer
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 4.0f;
+        } else {
+            return; // wait until back on floor
+        }
+    } else {
+        if ((this->actor.bgCheckFlags & 1)) { // about to hit the floor
```

Comment:
```suggestion
        if (this->actor.bgCheckFlags & 1) {
```
Same as comment above